### DISCUSSION TITLE
Order by 优化

### DISCUSSION THREAD

**Discussion by Wilson-BT** [apache/doris#13658] [RFC]
作为ap数据库，order by ... limit ... 的性能是否应该得进一步优化一下，比如order by 某个字段的时候，时长会随着select 出来的字段增多，执行时间指数型增长。此时是否应该只对 order by 的字段进行排序，而不用对所有select 出来的字段进行排序？比如当我们scan 所有字段出来，order by 之前转为行？再进行排序？

**Comment by HappenLee** [comment]
`order by` is supposed to sort only on specified fields, not on all fields by `select` ?

**Reply by Wilson-BT** [reply]
但是，为啥 select * from xxx order by id limit 10和select id  from xxx  order by id limit 10 的执行时间差异会这么大呢，select 1 个字段 和select 5 个字段速度差距为两倍
